internal/stores/hostdb.go

```diff
@@ -737,7 +741,15 @@ func (ss *SQLStore) RecordInteractions(ctx context.Context, interactions []hostd
 						return err
 					}
 					host.Settings = convertHostSettings(sr.Settings)
-					host.PriceTable = convertHostPriceTable(sr.PriceTable)
+
+					// only update the price table if it is newer
+					//
+					// TODO: this can eventually be smarter and select the new
+					// price table if its prices are better
+					if sr.PriceTableExpiry.After(host.PriceTableExpiry) {
```

Comment:
I'm refactoring it so I will resolve this comment but the idea is that scans only update the price table on the host entity if the current price table is expired. That is because price tables for scans are not valid, so we never want to overwrite a paid for price table with one that we got from a scan.